controllers/api.controllers.js

```diff
@@ -53,17 +61,43 @@ exports.postNewComment = (req, res, next) => {
   const { article_id } = req.params;
   const { username, body } = req.body;
 
-  if (username === undefined) {
-    return next({ status: 400, msg: "username is required" });
+  selectArticleById(article_id)
+    .then((article) => {
+      postCommentOnArticle(article_id, username, body)
```

Comment:
nesting of .thens()! You need to return your postCommentOnArticle.  Then you can chain on a promise to the .then it is in to access the it's resolution.  This will remove the need for multiple .catch blocks